src/parser/parser.ts

```diff
@@ -0,0 +1,13 @@
+/**
+ * @license
+ * Copyright (c) 2016 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+ */
+
+export interface Parser<T> {
+  parse(contents: string, url: string): T;
```

Comment:
You know, if this returned `Promise<T>` we could use something like [webworker-threads](https://www.npmjs.com/package/webworker-threads) to do parallel parsing in a thread pool. It's not unlikely that parsing will be the slowest part of most analysis runs, so it may be worth the effort, or at least not shutting the door on the possibility.